File: UAS-UTS_Aplikasi Mobile/app/src/main/java/com/example/utsmobile/MainActivity.java
package com.example.utsmobile;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.media.Image;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private ImageButton androidImageButton3;
    private ImageButton androidImageButton2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        androidImageButton3 = (ImageButton) findViewById(R.id.imageButton3);
        androidImageButton2 = (ImageButton) findViewById(R.id.imageButton2);

        androidImageButton3.setOnClickListener(new View.OnClickListener()
        {
           public void onClick(View v){
               openLogin();
           }
        });
        androidImageButton2.setOnClickListener(new View.OnClickListener()
        {
            public void onClick(View v){
                openRegistrasi();
            }
        });
    }

    public void openLogin() {
        Intent intent = new Intent(this, Login.class);
        startActivity(intent);
    }
    public void openRegistrasi() {
        Intent intent = new Intent(this, Registrasi.class);
        startActivity(intent);
    }

}

File: source_code/VapingInfo/app/src/main/java/com/study/vapinginfo/MainActivity.kt
package com.study.vapinginfo

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    private lateinit var rvVapor: RecyclerView
    private var list:ArrayList<Vapor> = arrayListOf()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        rvVapor = findViewById(R.id.vpVaping)
        rvVapor.setHasFixedSize(true)

        list.addAll(VaporData.listData)
        showRecyclerList()
    }

    private fun showRecyclerList(){
        rvVapor.layoutManager = LinearLayoutManager(this)
        val vaporAdapter = VaporAdapter(list, this)
        rvVapor.adapter = vaporAdapter
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.menu_profile -> {
                val profileIntent = Intent(this, ProfileActivity::class.java)
                startActivity(profileIntent)
            }
        }
        return super.onOptionsItemSelected(item)
    }
}
